ProcessLib/Parameter/MeshNodeParameter.h

```diff
@@ -0,0 +1,59 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESSLIB_MESHNODEPARAMETER_H
+#define PROCESSLIB_MESHNODEPARAMETER_H
+
+#include "Parameter.h"
+
+namespace MeshLib
+{
+template <typename T>
+class PropertyVector;
+}  // MeshLib
+
+namespace ProcessLib
+{
+/// A parameter represented by a mesh property vector.
+template <typename T>
+struct MeshNodeParameter final : public Parameter<T> {
+    MeshNodeParameter(MeshLib::PropertyVector<T> const& property)
+        : _property(property)
+        , _cache(_property.getNumberOfComponents())
+    {
+    }
+
+    unsigned getNumberOfComponents() const override
+    {
+        return _property.getNumberOfComponents();
+    }
+
+    std::vector<T> const& getTuple(double const /*t*/,
+                                   SpatialPosition const& pos) const override
+    {
+        auto const n = pos.getNodeID();
+        assert(n);
```

Comment:
Just a discussion idea: if getNodeID  is called, than a non-empty optional is expected (asserted). Maybe better to move the assertion into getNodeID and (if required) add an additional getOptionalNodeID() function.